.github/workflows/apero-docs-preview.yaml

```diff
@@ -0,0 +1,42 @@
+on:
+  push:
+    branches:
+      - gha
+
+name: build-site
+
+jobs:
+  build-site:
+    runs-on: macOS-latest
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+      
+      - uses: r-lib/actions/setup-r@v1
+
+      - name: Install dependencies
+        run: |
+          install.packages("blogdown")
+          blogdown::install_hugo("0.79.0", extended=TRUE, force=TRUE)
+          blogdown::install_theme(theme="hugo-apero/hugo-apero", force=TRUE)
+        shell: Rscript {0}
+
+      - name: Build website
+        run: |
+          git config --local user.email "actions@github.com"
+          git config --local user.name "GitHub Actions"
+          Rscript -e 'blogdown::build_site(baseURL = "https://hugo-apero.github.io/apero/")'
+
+      - name: Add nojekyll
+        run: |
+          cd public/
+          touch .nojekyll
+          
+      - name: Deploy 🚀
+        uses: JamesIves/github-pages-deploy-action@3.6.2
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          BRANCH: gh-pages # The branch the action should deploy to.
+          FOLDER: public # The folder the action should deploy.
+          CLEAN: true # Automatically remove deleted files from the deploy branch
```

Comment:
Don't know if you tried several actions but I was using this one 
https://github.com/peaceiris/actions-gh-pages

which I find include more features - it will for example create the `.nojekyll` file by default and remove all files.

Just sharing. If it is working, it is great !

Reply:
I'll try those again, for some reason when I strayed to those actions I had a lot of error messages, so I didn't try harder and just aborted :) Was using all this as a useful distraction for myself!